internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -179,7 +189,37 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 			},
 		},
 
+		"scaling": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: false,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"capacity": {
+						Type:         pluginsdk.TypeInt,
+						Required:     true,
+						ValidateFunc: validation.IntPositive,
+					},
+				},
+			},
+		},
```

Comment:
Given there is only one property in this block, I think it would make sense here to simplify things by having just a top-level property called `capacity`

Reply:
Hi @catriona-m , this property is something there is active development on for the service. We expect to be extending this API to contain configuration properties for autoscaling in the near term 😄 

Is it preferred to change the property once there is a need for a nested set of properties and keep it simple now? I would assume that would later introduce a breaking change that would not be optimal.

Reply:
#### Flattening option:
```terraform
resource "azurerm_nginx_deployment" {
   // other stuff

   capacity = 40
   // OR (mutually exclusive)
   autoscaleProfile {
      name = "default"
      min = 40
      max = 4000
   }
   autoscaleProfile {
      // near term only one is supported, long term plans to be similar to  
      // https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_autoscale_setting#profile
   }
}
```

That would expand to ARM API call with `properties.scalingProperties.autoscaleProfile[0].{min: 40, max:4000, name:"default"}`

#### Nesting option

```terraform
resource "azurerm_nginx_deployment" {
   // other stuff
   
   scaling {
      capacity = 40
      // OR (mutually exclusive)
      autoscaleProfile {
         name = "daytime"
         capacity {
            min = 40
            max = 4000
         }
         time_window        = "PT5M"
      }
      autoscaleProfile {
        // near term only allows one autoscaleProfile, long term similar to 
        // https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_autoscale_setting#profile
      }
   }
   scaling {
     // whups this is an error
   }
}
```

Reply:
I like the flattening option.  I think this lines up with other azurerm terraform resources more helps guide towards correct usage.

Reply:
Agreed! I’ll work on those changes now.